### PR TITLE
create application configuration

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
It is not perfect (especially the infrastructure tests), but it's a start.
Let's build on top of it.

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -15,10 +15,8 @@
     ],
     "env": {
         "PG_POOL_MAX_SIZE": "3",
-        "ROCKET_ADDRESS": "0.0.0.0",
```
Comment (Bernardstanislas) [inline app.json:18]: hihi

File: app.json [hunk 2]
```diff
@@ -15,10 +15,8 @@
...
         "ROCKET_CLI_COLORS": "false",
-        "RUST_LOG": "debug",
-        "SLOG_CHANNEL_SIZE": "1024"
```
Comment (Bernardstanislas) [inline app.json:21]: Did you remove these variables directly in the Heroku apps too?

File: infrastructure/src/config.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	use super::*;
+
+	#[derive(Deserialize)]
+	pub struct App {
+		pub api_key: String,
+	}
+
+	#[test]
+	fn config_from_file() {
+		figment::Jail::expect_with(|jail| {
```
Comment (Bernardstanislas) [inline infrastructure/src/config.rs]: 🤯

File: api/src/app.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+local:
+  database:
+    url: postgres://postgres:postgres@localhost/marketplace_db
+    pool_max_size: 20
+
+staging:
```
[the remark this answers is not in the document]
Comment (Bernardstanislas) [reply api/src/app.yaml]: I mean it's a good sign to see "production" profile in a staging app config for example

File: api/src/app.yaml [hunk 5]
```diff
@@ -4,17 +4,26 @@ local:
...
+    admin_secret: $HASURA_GRAPHQL_ADMIN_SECRET
 
 staging:
   database:
     url: $DATABASE_URL
     pool_max_size: 3
   amqp:
     url: $CLOUDAMQP_URL
+  graphql:
+    url: https://staging.hasura.onlydust.xyz/v1/graphql
```
Comment (Bernardstanislas) [inline api/src/app.yaml]: Here for instance, one could argue that the distinction between staging and production is necessary to make them point at different endpoints. I would say that this means this endpoint should live in an env var, as stated by https://12factor.net/config

File: infrastructure/src/database/config.rs [hunk 6]
```diff
@@ -1,7 +1,8 @@
 use derive_getters::Getters;
+use derive_more::Constructor;
 use serde::Deserialize;
 
-#[derive(Deserialize, Getters)]
+#[derive(Deserialize, Getters, Constructor)]
```
Comment (Bernardstanislas) [inline infrastructure/src/database/config.rs:5]: For my own culture, what does it do?

File: app.json (hunk 2 above)
Comment (AnthonyBuisset) [reply app.json:21]: not yet, I will cleanup once code is loaded

File: api/src/app.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply api/src/app.yaml]: there could be different settings, for instance the graphql server URL is different

File: infrastructure/src/database/config.rs (hunk 6 above)
Comment (AnthonyBuisset) [reply infrastructure/src/database/config.rs:5]: creates the function `new` for you ❤️

File: api/src/app.yaml (hunk 5 above)
Comment (AnthonyBuisset) [reply api/src/app.yaml]: fair point, I can change that
this would also allow to put the `PROFILE` var in the `app.json`

**Review by Bernardstanislas — APPROVED** [review]

**Pull request merged**